### PR TITLE
[CHORE] [Integration test] Create test cases for survey detail page

### PR DESCRIPTION
**Pull request by manh-t**
## What happened 👀

Create test cases for survey detail page

## Proof Of Work 📹

https://user-images.githubusercontent.com/60863885/127426347-bd912e0e-ef76-416d-a6f9-a1fc2ffb52d6.mp4

```json
➜  fvm flutter drive --driver=test_driver/integration_test.dart  --target=integration_test/survey_detail_test.dart --flavor staging
Running "flutter pub get" in flutter_survey...                     996ms
Running Gradle task 'assembleStagingDebug'...                      21.6s
✓  Built build/app/outputs/flutter-apk/app-staging-debug.apk.
Installing build/app/outputs/flutter-apk/app.apk...                13.2s
VMServiceFlutterDriver: Connecting to Flutter application at http://127.0.0.1:58105/XRYFS6KFmLk=/
VMServiceFlutterDriver: Isolate found with number: 1794012601965191
VMServiceFlutterDriver: Isolate is paused at start.
VMServiceFlutterDriver: Attempting to resume isolate
D/EGL_emulation(30493): eglMakeCurrent: 0xe7564740: ver 3 0 (tinfo 0xe8735750)
D/eglCodecCommon(30493): setVertexArrayObject: set vao to 0 (0) 1 0
I/flutter (30493): 00:00 +0: Survey detail Page (setUpAll)
VMServiceFlutterDriver: Connected to Flutter application.
I/flutter (30493): 00:00 +1: Survey detail Page When navigating to survey detail page, it displays corresponding info
I/flutter (30493): 00:04 +2: Survey detail Page When tapping on Back button, it shows Dashboard page
I/flutter (30493): 00:08 +3: Survey detail Page When tapping on Start Survey button, it shows Take survey page
I/flutter (30493): 00:11 +3: Survey detail Page When navigating to survey detail page, it displays corresponding info
I/flutter (30493): This is fake
I/flutter (30493): ══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
I/flutter (30493): The following UnimplementedError was thrown running a test (but after the test had completed):
I/flutter (30493): UnimplementedError
I/flutter (30493): 
I/flutter (30493): When the exception was thrown, this was the stack:
I/flutter (30493): #0      FakeSurveyService.getSurveyDetail (package:flutter_survey/api/mock_service/fake_survey_service.dart:12:5)
I/flutter (30493): #1      SurveyRepositoryImpl.getSurveyDetail (package:flutter_survey/api/repository/survey_repository.dart:43:27)
I/flutter (30493): #2      GetSurveyDetailUseCase.call (package:flutter_survey/usecase/get_survey_detail_use_case.dart:19:30)
I/flutter (30493): #3      TakeSurveyModel.init (package:flutter_survey/pages/takesurvey/take_survey_model.dart:33:29)
I/flutter (30493): #4      _TakeSurveyPageState.initState.<anonymous closure> (package:flutter_survey/pages/takesurvey/take_survey_page.dart:32:54)
I/flutter (30493): (elided 16 frames from class _RawReceivePortImpl, class _Timer, dart:async, dart:async-patch, and package:stack_trace)
I/flutter (30493): ════════════════════════════════════════════════════════════════════════════════════════════════════
I/flutter (30493): 00:15 +4: Survey detail Page (tearDownAll)
I/flutter (30493): 00:15 +5: (tearDownAll)
I/flutter (30493): 00:15 +6: All tests passed!
All tests passed.
```

### CODE REVIEW

**Review by markgravity — APPROVED** [review]
LGTM 🤖

File: integration_test/survey_detail_test.dart [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+          Routes.TAKE_SURVEY_PAGE: (BuildContext context) => TakeSurveyPage(),
+        },
+      ));
+      await tester.pumpAndSettle();
+
+      await tester.tap(find.byKey(DashboardPageKey.nextButton));
+      await tester.pumpAndSettle();
+
+      await tester.tap(find.byKey(SurveyDetailPageKey.startSurveyButton));
+      await tester.pump(Duration(seconds: 4));
```
Comment (sleepylee) [inline integration_test/survey_detail_test.dart:82]: bump 4 sec? 😳  do we really need it that much just curious, @manh-t

Comment (manh-t) [reply integration_test/survey_detail_test.dart:82]: omg 😱 I don't think we need to pump to 4 secs here. Let me recheck it

**Comment by manh-t** [comment]
I reduced the pump duration, please help me review again @sleepylee 😓 🙏

**Review by sleepylee — APPROVED** [review]

**Pull request merged**